src/modules/ekf2/EKF2.cpp

```diff
@@ -1553,27 +1553,33 @@ void EKF2::UpdateMagSample(ekf2_timestamps_s &ekf2_timestamps)
 
 void EKF2::UpdateRangeSample(ekf2_timestamps_s &ekf2_timestamps)
 {
-	if (!_distance_sensor_selected) {
-		// get subscription index of first downward-facing range sensor
-		uORB::SubscriptionMultiArray<distance_sensor_s> distance_sensor_subs{ORB_ID::distance_sensor};
-
-		for (unsigned i = 0; i < distance_sensor_subs.size(); i++) {
-			distance_sensor_s distance_sensor;
-
-			if (distance_sensor_subs[i].copy(&distance_sensor)) {
-				// only use the first instace which has the correct orientation
-				if ((hrt_elapsed_time(&distance_sensor.timestamp) < 100_ms)
-				    && (distance_sensor.orientation == distance_sensor_s::ROTATION_DOWNWARD_FACING)) {
-
-					if (_distance_sensor_sub.ChangeInstance(i)) {
-						PX4_INFO("%d - selected distance_sensor:%d", _instance, i);
-						_distance_sensor_selected = true;
-					}
-				}
+	// get subscription index of first downward-facing range sensor
+	uORB::SubscriptionMultiArray<distance_sensor_s> distance_sensor_subs {
```

Comment:
We can structure this a bit differently if we're going to be checking all available instances constantly. Subscribing every iteration like this is still more expensive than it needs to be.

Let me take a look and propose something specific.